docs/how-to/virtual_memory.rst

```diff
@@ -0,0 +1,150 @@
+larger buffer allocated, than you initially need, which isn't an efficient way to handle
+resources and doesn't solve the issue of reallocation, when the extra buffer runs out.
+
+Virtual memory management is the solution to these memory management problems. It helps
+to reduce memory usages and unnecessary ``memcpy`` calls.
+
+.. _memory_allocation_virtual_memory:
+Memory allocation
+===============================================================================
+
+When you need a chunk of memory on the device, previously you called the ``hipMalloc``
+function to allocate and it returns a pointer, that is immediately usable after the call.
+When using virtual memory this is separated into multiple steps using the
+``hipMemCreate``, ``hipMemAddressReserve``, ``hipMemMap`` and ``hipMemSetAccess``
+functions. In this guide you'll learn what these functions do and how you can use them
+for your needs.
+
+Allocate physical memory
+-------------------------------------------------------------------------------
+
+The first step is allocate the physical memory itself with the ``hipMemCreate`` function.
+It take the size of the buffer, an ``unsigned long long`` for flags and a
+``hipMemAllocationProp``, which contains the properties of the properties of the of the
+memory to be allocated, like where is this memory physically located or what kind of
+shareable handles should be available. If the allocation is successful the function
+returns with a ``hipSuccess`` and the ``hipMemGenericAllocationHandle_t`` will represent
+a valid physical memory allocation. It is important to note that the size has to be
+aligned with the granularity appropriate for the properties of the allocation. You can
+use the ``hipMemGetAllocationGranularity`` function to query this information.
+
+.. code-block::cpp
+
+    size_t granularity = 0;
+    hipMemGenericAllocationHandle_t allocHandle;
+    hipMemAllocationProp prop = {};
+    prop.type = HIP_MEM_ALLOCATION_TYPE_PINNED;
+    prop.location.type = HIP_MEM_LOCATION_TYPE_DEVICE;
+    prop.location.id = currentDev;
+    hipMemGetAllocationGranularity(&granularity, &prop, HIP_MEM_ALLOC_GRANULARITY_MINIMUM);
+    padded_size = ROUND_UP(size, granularity);
+    hipMemCreate(&allocHandle, padded_size, &prop, 0);
+
+Reserve virtual address range
+-------------------------------------------------------------------------------
+
+Once you have a physical memory allocation you need to map it, before you can use it and
+you need a virtual address to map to. To reserve a virtual memory range use the
+``hipMemAddressReserve`` function, which needs the same size as the physical memory
+allocate previously. Afterwards you can map physical memory allocation to the newly
+acquired virtual memory address range with the ``hipMemMap`` function. Mapping means that
+the physical allocation will be available through the virtual address range it is mapped
+to.
+
+.. code-block::cpp
```

Comment:
Same problem with this code block. They all have this problem.

Reply:
Fixed.